evap/evaluation/models.py

```diff
@@ -284,7 +287,11 @@ def can_user_see_results(self, user):
         if user.is_staff:
             return True
         if self.state == 'published':
-            return self.can_publish_grades or self.is_user_contributor_or_delegate(user)
+            if self.is_user_contributor_or_delegate(user):
+                return True
+            if not self.can_publish_grades:
+                return False
+            return not self.is_private or user in self.participants.all()
```

Comment:
i thought about `return False` as default but didnt find something nice. i thought True as default would be ok since we're in state published, so everything is published by default except for special cases.

False as default might look like this:
```
if self.state == 'published':
    if self.is_user_contributor_or_delegate(user):
        return True
    if self.can_publish_grades:
        if not self.is_private or user in self.participants.all():
            return True
    return False
return False
```
i think that might be better. your call :)